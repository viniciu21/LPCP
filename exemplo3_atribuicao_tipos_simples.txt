declaration:
    a: int; 
    b: string;
    c: char;
    d: float;
    e: bool;
end_declaration

main:
    a = 10;
    b = "oi";
    c = 'o';
    d = 10.102;
    e = false;
end_main