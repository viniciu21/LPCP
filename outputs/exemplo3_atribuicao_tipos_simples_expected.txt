Declaração de variável: Id "a" (2,5)(True,[(Id "a" (2,5),IntValue 0 (2,8))],[],[],[])
Declaração de variável: Id "b" (3,5)(True,[(Id "a" (2,5),IntValue 0 (2,8)),(Id "b" (3,5),StringValue "" (3,8))],[],[],[])
Declaração de variável: Id "c" (4,5)(True,[(Id "a" (2,5),IntValue 0 (2,8)),(Id "b" (3,5),StringValue "" (3,8)),(Id "c" (4,5),CharValue '\NUL' (4,8))],[],[],[]) 
Declaração de variável: Id "d" (5,5)(True,[(Id "a" (2,5),IntValue 0 (2,8)),(Id "b" (3,5),StringValue "" (3,8)),(Id "c" (4,5),CharValue '\NUL' (4,8)),(Id "d" (5,5),FloatValue 0.0 (5,8))],[],[],[])
Declaração de variável: Id "e" (6,5)(True,[(Id "a" (2,5),IntValue 0 (2,8)),(Id "b" (3,5),StringValue "" (3,8)),(Id "c" (4,5),CharValue '\NUL' (4,8)),(Id "d" (5,5),FloatValue 0.0 (5,8)),(Id "e" (6,5),BoolValue False (6,8))],[],[],[])
Atualização de estado sobre a variável: Id "a" (10,5)(True,[(Id "a" (2,5),IntValue 10 (10,9)),(Id "b" (3,5),StringValue "" (3,8)),(Id "c" (4,5),CharValue '\NUL' (4,8)),(Id "d" (5,5),FloatValue 0.0 (5,8)),(Id "e" (6,5),BoolValue False (6,8))],[],[],[])
Atualização de estado sobre a variável: Id "b" (11,5)(True,[(Id "a" (2,5),IntValue 10 (10,9)),(Id "b" (3,5),StringValue "oi" (11,9)),(Id "c" (4,5),CharValue '\NUL' (4,8)),(Id "d" (5,5),FloatValue 0.0 (5,8)),(Id "e" (6,5),BoolValue False (6,8))],[],[],[])
Atualização de estado sobre a variável: Id "c" (12,5)(True,[(Id "a" (2,5),IntValue 10 (10,9)),(Id "b" (3,5),StringValue "oi" (11,9)),(Id "c" (4,5),CharValue 'o' (12,9)),(Id "d" (5,5),FloatValue 0.0 (5,8)),(Id "e" (6,5),BoolValue False (6,8))],[],[],[])
Atualização de estado sobre a variável: Id "d" (13,5)(True,[(Id "a" (2,5),IntValue 10 (10,9)),(Id "b" (3,5),StringValue "oi" (11,9)),(Id "c" (4,5),CharValue 'o' (12,9)),(Id "d" (5,5),FloatValue 10.102 (13,9)),(Id "e" (6,5),BoolValue False (6,8))],[],[],[])
Atualização de estado sobre a variável: Id "e" (14,5)(True,[(Id "a" (2,5),IntValue 10 (10,9)),(Id "b" (3,5),StringValue "oi" (11,9)),(Id "c" (4,5),CharValue 'o' (12,9)),(Id "d" (5,5),FloatValue 10.102 (13,9)),(Id "e" (6,5),BoolValue False (14,9))],[],[],[])
Tokens do programa: [Declaration (1,1),Colon (1,12),Id "a" (2,5),Colon (2,6),Type "int" (2,8),SemiColon (2,11),Id "b" (3,5),Colon (3,6),Type "string" (3,8),SemiColon (3,14),Id "c" (4,5),Colon (4,6),Type "char" (4,8),SemiColon (4,12),Id "d" (5,5),Colon (5,6),Type "float" (5,8),SemiColon (5,13),Id "e" (6,5),Colon (6,6),Type "bool" (6,8),SemiColon (6,12),EndDeclaration (7,1),Main (9,1),Colon (9,5),Id "a" (10,5),Assign (10,7),IntValue 10 (10,9),SemiColon (10,11),Id "b" (11,5),Assign (11,7),StringValue "oi" (11,9),SemiColon (11,13),Id "c" (12,5),Assign (12,7),CharValue 'o' (12,9),SemiColon (12,12),Id "d" (13,5),Assign (13,7),FloatValue 10.102 (13,9),SemiColon (13,15),Id "e" (14,5),Assign (14,7),BoolValue False (14,9),SemiColon (14,14),EndMain (15,1)]